projects/opentrain/gtfs_pipeline/heatmap/HeatMap.java

```diff
@@ -0,0 +1,540 @@
+    Stop stop;
+    double arrivalTime;
+
+    public StopsAround(StopTime stopTime, double arrivalTime) {
+
+      this.arrivalTime = arrivalTime;
+      setStop(stopTime);
+    }
+
+    private void setStop(StopTime stopTime) {
+      // From all stops class pick the chosen stop from stopsTime
+      // *to use longitude/latitude
+      for (Stop stopAround : stopsList) {
+
+        if (stopAround.getStopId() == stopTime.getStopId()) {
+          stop = stopAround;
+          break;
+        }
+      }
+    }
+  }
+
+  // arrivalTime To Location by stops.latitude,stops.longitude make matrix of colors
+  private void arrivalTimeToLocation() {
+    double maxLongitude = 0;
+    double maxLatitude = 0;
+
+    for (StopsAround stopAround : stopsAroundList) {
+      if (stopAround.stop.getStopLat() > maxLatitude) {
+        maxLatitude = (stopAround.stop.getStopLat());
+      }
+
+      if (stopAround.stop.getStopLon() > maxLongitude) {
+        maxLongitude = stopAround.stop.getStopLon();
+      }
+    }
+    // maxLongitude = 35.23; //Max Longitude point on map(in the borders)
+    // maxLatitude = 32.75; //Max Latitude point on map(in the borders)
+
+    // Location matrix *100 to make difference between points
+    // *2 to up and down, *2 right and left
+    heatMap = new double[(int) (maxLatitude * 100) * 2][(int) (maxLongitude * 100) * 2];
+
+    for (StopsAround stopAround : stopsAroundList) {
+
+      // Arrival time of each stop from the chosen stop by location
+      // *100 to make difference between locations
+      stopAround.i = (int) (stopAround.stop.getStopLat() * 100);
+      stopAround.j = (int) (stopAround.stop.getStopLon() * 100);
+      heatMap[stopAround.i][stopAround.j] = (int) (stopAround.arrivalTime);
+    }
+
+    // Distance of other points to the nearest stop
+    for (int i = 0; i < heatMap.length; i++) {
+      for (int j = 0; j < heatMap[0].length; j++) {
+
+        if (heatMap[i][j] == 0) // not a stop
+        {
+          StopsAround firstStop = stopsAroundList.get(0);
+          double ac = Math.abs(0 - firstStop.i); // y distance
+          double cb = Math.abs(0 - firstStop.j); // x distance
+          double distance = Math.hypot(ac, cb);
+          double stopTimeArrival = firstStop.arrivalTime;
+
+          // Find nearest stop
+          for (StopsAround stopAround : stopsAroundList) {
+
+            ac = Math.abs(i - stopAround.i); // y distance
+            cb = Math.abs(j - stopAround.j); // x distance
+
+            if (distance > Math.hypot(ac, cb)) {
+              distance = Math.hypot(ac, cb);
+              stopTimeArrival = stopAround.arrivalTime;
+            }
+          }
+          // *60 minutes to seconds
+          heatMap[i][j] = (int) ((distance * 60) + stopTimeArrival);
+        }
+        if (heatMap[i][j] == 0) {
+          System.out.println("Check heatMap assignment");
+        }
+      }
+    }
+  }
+
+  // Draw only the heat map colors
+  public void drawHeatMap(String imagePath) {
```

Comment:
`imagePath` parameter is never used. Do we need it?

Reply:
Yes, we can create two images once,
one that contains only the heat map colors
and the second contains the heat map on area map( Israel map).